test/tf_exec.cpp

```diff
@@ -42,6 +43,91 @@ namespace testing {
 
 #define ASSERT_OK(x) ASSERT_EQ((x), ::tensorflow::Status::OK());
 
+Status LoadGraph(const string& graph_file_name,
+                 std::unique_ptr<tensorflow::Session>* session,
+                 const tensorflow::SessionOptions& options) {
+  tensorflow::GraphDef graph_def;
+  auto load_graph_status =
+      ReadTextProto(Env::Default(), graph_file_name, &graph_def);
+  if (!load_graph_status.ok()) {
+    return tensorflow::errors::NotFound("Failed to load compute graph at '",
+                                        graph_file_name, "'");
+  }
+  session->reset(tensorflow::NewSession(options));
+  Status session_create_status = (*session)->Create(graph_def);
+  if (!session_create_status.ok()) {
+    return session_create_status;
+  }
+  return Status::OK();
+}
+
+Status CreateSession(const string& graph_filename,
+                     unique_ptr<Session>& session) {
+  SessionOptions options;
+  options.config.mutable_graph_options()
+      ->mutable_optimizer_options()
+      ->set_opt_level(OptimizerOptions_Level_L0);
+  options.config.mutable_graph_options()
+      ->mutable_rewrite_options()
+      ->set_constant_folding(RewriterConfig::OFF);
+
+  if (ngraph_tf_is_grappler_enabled()) {
+    auto* custom_config = options.config.mutable_graph_options()
+                              ->mutable_rewrite_options()
+                              ->add_custom_optimizers();
+
+    custom_config->set_name("ngraph-optimizer");
+    (*custom_config->mutable_parameter_map())["ngraph_backend"].set_s("CPU");
+    (*custom_config->mutable_parameter_map())["device_id"].set_s("0");
+
+    options.config.mutable_graph_options()
+        ->mutable_rewrite_options()
+        ->set_min_graph_nodes(-1);
+
+    options.config.mutable_graph_options()
+        ->mutable_rewrite_options()
+        ->set_meta_optimizer_iterations(RewriterConfig::ONE);
+  }
+
+  // Load the network
+  Status load_graph_status = LoadGraph(graph_filename, &session, options);
+  return load_graph_status;
+}
+
+TEST(tf_exec, SingleGraphOn2Threads) {
+  string graph_name = "test_axpy.pbtxt";
+
+  unique_ptr<Session> session;
+  ASSERT_OK(CreateSession(graph_name, session));
+
+  auto worker = [&session](size_t thread_id) {
+    string inp_tensor_name_0{"x"};
+    string inp_tensor_name_1{"y"};
+    string out_tensor_name{"add"};
+    std::vector<Tensor> out_tensor_vals;
+
+    Tensor inp_tensor_val(tensorflow::DT_FLOAT,
+                          tensorflow::TensorShape({2, 3}));
+    AssignInputValuesRandom<float>(inp_tensor_val, 0, 10);
+
+    std::vector<std::pair<string, tensorflow::Tensor>> inputs = {
+        {inp_tensor_name_0, inp_tensor_val},
+        {inp_tensor_name_1, inp_tensor_val}};
+
+    for (int i = 0; i < 10; i++) {
+      NGRAPH_VLOG(5) << "thread_id: " << thread_id << " started: " << i;
+      ASSERT_OK(session->Run(inputs, {out_tensor_name}, {}, &out_tensor_vals));
+      NGRAPH_VLOG(5) << "thread_id: " << thread_id << " finished: " << i;
+    }
+  };
```

Comment:
Should we be checking if the right values are being returned to the right thread or is that out of scope for this PR?

Reply:
Added comparision with expected values